numpy/ma/tests/test_core.py

```diff
@@ -2671,6 +2671,24 @@ def test_put(self):
         assert_array_equal(x, [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, ])
         assert_equal(x.mask, [1, 0, 0, 0, 1, 1, 0, 0, 0, 0])
 
+    def test_put_nomask(self):
+        # GitHub issue 6425
+        x = zeros(10)
+        z = array([3., -1.], mask=[False, True])
+        self.assertTrue(x[1] is not masked)
+        self.assertTrue(x[2] is not masked)
+        self.assertTrue(x[3] is not masked)
+        self.assertTrue(z[1] is masked)
```

Comment:
Are these 4 asserts needed? If they are only testing basic MA functionality which is tested elsewhere, I favor removing them for clarity.

The asserts below are good and make clear what we are testing.